### PR TITLE
[Feature] Support tabular's signed s3 temp credentials

### PR DESCRIPTION
**Pull request by Smith-Cruise**
After this pr, we can using tabular by below way:
```sql
create external catalog tabular properties (
  "type"="iceberg",
  "iceberg.catalog.type"="rest",
  "iceberg.catalog.uri"="https://api.tabular.io/ws",
  "iceberg.catalog.credential"="t-5Ii8e3FIbT9m0",
  "iceberg.catalog.warehouse"="sandbox"
);
```
You don't need to configure aws s3 credentials anymore, we will configure it automatically by the tabular signer.

> You can create db, table and insert values like normal iceberg catalog.

## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 54 (11.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/rest/IcebergRESTCatalog.java|0|5|00.00%|[75, 76, 83, 84, 85]|
|:large_blue_circle:|com/starrocks/credential/CloudConfigurationFactory.java|0|15|00.00%|[66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 83]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|0|9|00.00%|[384, 387, 388, 389, 390, 392, 393, 394, 395]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/IcebergAwsClientFactory.java|0|7|00.00%|[96, 107, 149, 179, 205, 206, 208]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudCredential.java|3|12|25.00%|[146, 148, 149, 151, 191, 192, 193, 196, 235]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudConfigurationFactory.java|1|2|50.00%|[69]|
|:large_blue_circle:|com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java|2|4|50.00%|[138, 189]|

File: be/src/fs/credential/cloud_configuration_factory.h [hunk 1]
```diff
@@ -29,6 +29,7 @@ static const std::string AWS_S3_USE_AWS_SDK_DEFAULT_BEHAVIOR = "aws.s3.use_aws_s
 static const std::string AWS_S3_USE_INSTANCE_PROFILE = "aws.s3.use_instance_profile";
 static const std::string AWS_S3_ACCESS_KEY = "aws.s3.access_key";
 static const std::string AWS_S3_SECRET_KEY = "aws.s3.secret_key";
+static const std::string AWS_S3_SESSION_TOKEN = "aws.s3.session_token";
```
Comment (stephen-shelby) [inline be/src/fs/credential/cloud_configuration_factory.h:32]: add some comments about this tabular properties.

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java [hunk 2]
```diff
@@ -132,7 +133,12 @@ public void initialize(Map<String, String> properties) {
             switch (type.toLowerCase(Locale.ROOT)) {
                 case "hive":
                 case "rest":
-                    wrappedIO = new HadoopFileIO(conf);
+                    if (properties.getOrDefault(KEY_ENABLE_TABULAR_SUPPORT, "false").equals("true")) {
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java:136]: It's better to optimize this block for legibility

File: be/src/fs/credential/cloud_configuration_factory.h (hunk 1 above)
Comment (Smith-Cruise) [reply be/src/fs/credential/cloud_configuration_factory.h:32]: session token is not only for tabular, it's aws temporary ak/sk.
That means we can use below grammar now:
```sql
"aws.s3.access_key"="xxx",
"aws.s3.secret_key"="yyy",
"aws.s3.session_token"="xxx"
```

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java (hunk 2 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java]: is there any good advice?

File: fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationFactory.java [hunk 3]
```diff
@@ -61,6 +63,27 @@ public static CloudConfiguration buildCloudConfigurationForStorage(Map<String, S
         return buildDefaultCloudConfiguration();
     }
 
+    public static CloudConfiguration buildCloudConfigurationForTabular(Map<String, String> properties) {
+        Map<String, String> copiedProperties = new HashMap<>();
+        String tmpAk = properties.getOrDefault(AwsProperties.S3FILEIO_ACCESS_KEY_ID, null);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationFactory.java:68]: sessionAk ？

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergTableSink.java [hunk 4]
```diff
@@ -74,7 +76,16 @@ public IcebergTableSink(IcebergTable icebergTable, TupleDescriptor desc, boolean
         Connector connector = GlobalStateMgr.getCurrentState().getConnectorMgr().getConnector(catalogName);
         Preconditions.checkState(connector != null,
                 String.format("connector of catalog %s should not be null", catalogName));
-        this.cloudConfiguration = connector.getCloudConfiguration();
+
+        // Try to set for tabular
+        CloudConfiguration tabularTempCloudConfiguration = CloudConfigurationFactory.
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/planner/IcebergTableSink.java:81]: You can consider checking if the 'tabular' option is available first and then build the cloud configuration accordingly.

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/InsertPlanTest.java [hunk 5]
```diff
@@ -790,7 +795,38 @@ public void testInsertIcebergTableSink() throws Exception {
         starRocksAssert.withCatalog(createIcebergCatalogStmt);
         MetadataMgr metadata = starRocksAssert.getCtx().getGlobalStateMgr().getMetadataMgr();
 
-        Table nativeTable = new BaseTable(null, null);
+        Table nativeTable = new BaseTable(new TableOperations() {
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/InsertPlanTest.java:798]: you can mock `io()` method.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 17 / 56 (30.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/rest/IcebergRESTCatalog.java|0|3|00.00%|[82, 83, 84]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|0|8|00.00%|[368, 371, 372, 373, 375, 376, 377, 378]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/IcebergAwsClientFactory.java|0|7|00.00%|[96, 107, 149, 179, 205, 206, 208]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudCredential.java|3|12|25.00%|[146, 148, 149, 151, 191, 192, 193, 196, 235]|
|:large_blue_circle:|com/starrocks/credential/CloudConfigurationFactory.java|7|15|46.67%|[73, 74, 75, 76, 77, 78, 79, 80]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudConfigurationFactory.java|1|2|50.00%|[69]|
|:large_blue_circle:|com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java|2|4|50.00%|[139, 190]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|4|5|80.00%|[84]|

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/InsertPlanTest.java (hunk 5 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/test/java/com/starrocks/sql/plan/InsertPlanTest.java]: fixed

File: fe/fe-core/src/main/java/com/starrocks/planner/IcebergTableSink.java (hunk 4 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/planner/IcebergTableSink.java]: fixed

File: fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationFactory.java (hunk 3 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/credential/CloudConfigurationFactory.java]: changed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 58 (13.79%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/rest/IcebergRESTCatalog.java|0|3|00.00%|[82, 83, 84]|
|:large_blue_circle:|com/starrocks/credential/CloudConfigurationFactory.java|0|15|00.00%|[67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 84]|
|:large_blue_circle:|com/starrocks/planner/IcebergScanNode.java|0|8|00.00%|[368, 371, 372, 373, 375, 376, 377, 378]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/IcebergAwsClientFactory.java|0|7|00.00%|[96, 107, 149, 179, 205, 206, 208]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudCredential.java|3|12|25.00%|[146, 148, 149, 151, 191, 192, 193, 196, 235]|
|:large_blue_circle:|com/starrocks/planner/IcebergTableSink.java|2|7|28.57%|[83, 84, 86, 88, 89]|
|:large_blue_circle:|com/starrocks/credential/aws/AWSCloudConfigurationFactory.java|1|2|50.00%|[69]|
|:large_blue_circle:|com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java|2|4|50.00%|[139, 190]|

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-3.1

**Pull request merged**